### PR TITLE
Document pull from platform with multiple DBs

### PR DESCRIPTION
**Pull request by bserem**
## The Problem/Issue/Bug:

`ddev pull` won't work when multiple mysql databases are configured in a platformsh project (called `relationships`)

## How this PR Solves The Problem:

It doesn't, but provides documentation about it, which is equally good :)
A two command workaround are provided to achieve what `ddev pull` would do.

## Manual Testing Instructions:

Create a platformsh project with multiple mysql databases in relationships and follow the docs.

## Related Issue Link(s):

- #4415 
- https://github.com/platformsh/ddev-platformsh/issues/59

## Release/Deployment notes:

It's just documentation.

<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/4417"></a>

### CODE REVIEW

File: docs/content/users/providers/platform.md [hunk 1]
```diff
@@ -46,3 +46,30 @@ web_environment:
...
+```
+relationships:
+  database: 'mysqldb:mysql'
+  legacy: 'legacydb:mysql'
+  redis: 'rediscache:redis'
+```
+
+Once the download has finished, you'll need to import the DB to DDEV:
+```
+ddev import-db --src=.ddev/.downloads/db.sql.gz
```
[the remark this answers is not in the document]
Comment (bserem) [reply docs/content/users/providers/platform.md:72]: It will be imported in the default DB (named `db`). I am under the assumption one uses ddev with the defaults (so, one database).
I am not taking platformsh/ddev-platformsh into consideration at all.

To be honest, I haven't dealt with multi-db projects in ddev for several years now.

[the remark this answers is not in the document]
Comment (rfay) [reply docs/content/users/providers/platform.md:72]: So you're talking about the situation where there are multiple databases on Platform.sh but we only want to use one of them (as the default db)?

[the remark this answers is not in the document]
Comment (bserem) [reply docs/content/users/providers/platform.md:72]: Exactly! Like, for example, only one of them is required for Drupal.
I updated the changes to reflect multiple local databases too.

**Comment by bserem** [comment]
I should mention that research and work on this and #4415 is sponsored by [Annertech](annertech.com)

**Comment by rfay** [comment]
Thanks to you and to [Annertech](https://github.com/drud/ddev/pull/annertech.com)!

File: docs/content/users/providers/platform.md [hunk 2]
```diff
@@ -46,3 +46,34 @@ web_environment:
...
+
+## Edge Cases
+
+### Multiple databases in a project
+
+Platform.sh allows for multiple databases to be available in a project. This is done via [relationships](https://docs.platform.sh/create-apps/app-reference.html#relationships). When using multiple databases `ddev pull` won't work as it [doesn't know which database to pull](https://github.com/drud/ddev/issues/4415).
+
+To bypass the problem, you can use this command (from host):
+```
+ddev platform db:dump --yes --gzip --file=/var/www/html/.ddev/.downloads/db.sql.gz --project="${PLATFORM_PROJECT}" --environment="${PLATFORM_ENVIRONMENT}" --relationship RELATIONSHIP_NAME
```
Comment (rfay) [inline docs/content/users/providers/platform.md:59]:
```suggestion
ddev exec platform db:dump --yes --gzip --file=/var/www/html/.ddev/.downloads/db.sql.gz --project="${PLATFORM_PROJECT}" --environment="${PLATFORM_ENVIRONMENT}" --relationship <RELATIONSHIP_NAME>
```

`ddev platform` is not a standard ddev command, it's provided by the ddev-platformsh add-on. So we'd need to use `ddev exec platform` here.

**Review by rfay — CHANGES_REQUESTED** [review]
Overall, I think this would be better as an edited platform.yaml file. Remove the #ddev-generated from platform.yaml and adjust it for the project to have the right relationship. It could be called something other than platform.yaml as well. Maybe we should add comments to the platform.yaml about how to do this?

File: docs/content/users/providers/platform.md [hunk 3]
```diff
@@ -46,3 +46,34 @@ web_environment:
 
 * `ddev pull platform` will connect to Platform.sh to download database and files. To skip downloading and importing either file or database assets, use the `--skip-files` and `--skip-db` flags.
 * If you need to change the `platform.yaml` recipe, you can change it to suit your needs, but remember to remove the `#ddev-generated` line from the top of the file.
+
```
Comment (rfay) [inline docs/content/users/providers/platform.md:49]:
```suggestion
```

Fix docs test complaint

File: docs/content/users/providers/platform.md [hunk 4]
```diff
@@ -46,3 +46,34 @@ web_environment:
...
+
+## Edge Cases
+
+### Multiple databases in a project
+
+Platform.sh allows for multiple databases to be available in a project. This is done via [relationships](https://docs.platform.sh/create-apps/app-reference.html#relationships). When using multiple databases `ddev pull` won't work as it [doesn't know which database to pull](https://github.com/drud/ddev/issues/4415).
+
+To bypass the problem, you can use this command (from host):
```
Comment (rfay) [inline docs/content/users/providers/platform.md:57]:
```suggestion
To bypass the problem, you can use this command (from host):

```

Fix docs test complaint

File: docs/content/users/providers/platform.md [hunk 5]
```diff
@@ -46,3 +46,34 @@ web_environment:
...
+```
+ddev platform db:dump --yes --gzip --file=/var/www/html/.ddev/.downloads/db.sql.gz --project="${PLATFORM_PROJECT}" --environment="${PLATFORM_ENVIRONMENT}" --relationship RELATIONSHIP_NAME
+```
+
+To identify the name of the relationship look into you `.platform.app.yaml` file, left hand column is the name:
```
Comment (rfay) [inline docs/content/users/providers/platform.md:62]:
```suggestion
To identify the name of the relationship look into you `.platform.app.yaml` file, left hand column is the name:

```

Fix docs complaint

File: docs/content/users/providers/platform.md [hunk 6]
```diff
@@ -46,3 +46,34 @@ web_environment:
...
+```
+relationships:
+  database: 'mysqldb:mysql'
+  legacy: 'legacydb:mysql'
+  redis: 'rediscache:redis'
+```
+
+Once the download has finished, you'll need to import to the **default** DDEV DB (named `db`):
```
Comment (rfay) [inline docs/content/users/providers/platform.md:70]:
```suggestion
Once the download has finished, you'll need to import to the **default** DDEV DB (named `db`):

```

File: docs/content/users/providers/platform.md [hunk 7]
```diff
@@ -46,3 +46,34 @@ web_environment:
...
+```
+ddev import-db --src=.ddev/.downloads/db.sql.gz
+```
+
+If you also have a need for [more than one database in DDEV](../../basics/database-management) then adapt the above command as needed:
```
Comment (rfay) [inline docs/content/users/providers/platform.md:75]:
```suggestion
If you also have a need for [more than one database in DDEV](../../basics/database-management) then adapt the above command as needed:

```

File: docs/content/users/providers/platform.md [hunk 8]
```diff
@@ -46,3 +46,34 @@ web_environment:
...
+```
+ddev import-db --target-db=backend --src=.ddev/.downloads/backend.sql.gz
+```
+The above commands can easily be added in a `host` command, to facilitate things.
```
Comment (rfay) [inline docs/content/users/providers/platform.md:79]:
```suggestion

The above commands can easily be added in a `host` command, to facilitate things.
```

**Review by rfay — CHANGES_REQUESTED** [review]
Fix various docs test complaints.

**Comment by rfay** [comment]
@bserem could you take a look at 
* https://github.com/drud/ddev/pull/4426 as an alternative?

**Comment by rfay** [comment]
Closing in favor of https://github.com/drud/ddev/pull/4426, I know you'll be there to test that one. Thanks for raising this!

**Comment by bserem** [comment]
Sure thing! We'll revisit documentation once that one is done and merged.
